src/Features/Core/Portable/GenerateEqualsAndGetHashCodeFromMembers/AbstractGenerateEqualsAndGetHashCodeService.cs

```diff
@@ -147,6 +147,8 @@ public async Task<Document> FormatDocumentAsync(Document document, CancellationT
         {
             // If we have access to System.HashCode, then just use that.
             var hashCodeType = compilation.GetTypeByMetadataName("System.HashCode");
+            if (hashCodeType?.DeclaredAccessibility != Accessibility.Public)
```

Comment:
Should this be using one of the "is accessible" in this context APIs in case there's some bizarre InternalsVisibleTo going on or something? Or wouldn't this exclude somebody who has their own internal copy in a repo they can't upgrade to newer frameworks?

I usually wouldn't be terribly paranoid about scenarios like this but the customer feedback items implies that we actually might have these situations in the wild.

Reply:
fixing up.  i originally didn' tdo that because i wasn't in a location where that check was easy.  but it's easy enough here.